clang/lib/CIR/CodeGen/CIRGenModule.cpp

```diff
@@ -1656,9 +1656,11 @@ void CIRGenModule::emitGlobalDefinition(GlobalDecl gd, mlir::Operation *op) {
     if (const auto *method = dyn_cast<CXXMethodDecl>(d)) {
       // Make sure to emit the definition(s) before we emit the thunks. This is
       // necessary for the generation of certain thunks.
-      if (isa<CXXConstructorDecl>(method) || isa<CXXDestructorDecl>(method))
+      if (isa<CXXConstructorDecl>(method) || isa<CXXDestructorDecl>(method)) {
+        if (fd->getAttr<AnnotateAttr>())
+          deferredAnnotations[getMangledName(gd)] = cast<ValueDecl>(d);
```

Comment:
Are you sure you need to add this here? `EmitGlobal` both in OG and in the incubator handles it generically for FunctionDecl

Reply:
@xlauko ^

Reply:
Yea, I'm just pointing out how this work (as an answer to your question) via a rhetoric question. @Sharp0802 please remove this bit, if that doesn't do the job, you need to debug and find out what you are missing elsewhere.